src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -274,11 +274,21 @@ public void Dispatch()
                 }
```

Comment:
Am I reading the code correctly that we'll likely end up calling Set twice for a sync operation (with the second call just being a nop)?  I think that's probably fine, but it'd be worth a comment calling that out.

Reply:
Not sure if I missed something, why do you think it might called twice? My intention was that either the epoll thread handles an event or it will queue it for processing in the background, but not both. If the epoll thread correctly sees that there is a pending sync operation next, then it would call set and not queue an operation for that. Otherwise, it would not process the event and queue it instead.